server/plugin.ts

```diff
@@ -42,7 +43,7 @@ export class ObservabilityPlugin implements Plugin<ObservabilityPluginSetup, Obs
     core.http.registerRouteHandlerContext('observability_plugin', (context, request) => {
       return {
         logger: this.logger,
-        observabilityClient,
+        observabilityClient: OpenSearchObservabilityPlugin,
```

Comment:
This line replaces `observabilityClient` object with `OpenSearchObservabilityPlugin` in plugin context, I think what you want is to add `OpenSearchObservabilityPlugin` after `PPLPlugin` in line 39?

I don't think `observabilityClient` is being accessed through context anywhere so this doesn't break anything yet, but just want to double check if this is intentional @ps48

Reply:
Oh yes, that was intentional. The backend is not fully done so was waiting to finish the Opensearch Part and then add it here. But if for notebooks integration you want to add this, please go ahead.

Reply:
But why expose `OpenSearchObservabilityPlugin` in router context? it's a helper function to construct a client. caller needs to provide `Client` so it defines client actions in the Client prototype, i don't think you can use it directly?

I might need to modify this for notebooks to work, since current setup needs `context.observability_plugin.observabilityClient` to be a client, and I think we should only have one `observabilityClient`

Reply:
I think, I was about to change the name of Client object name [observabilityClient => openSearchObservabilityClient], but somehow forgot to do that. And yes, line 46 should be  `observabilityClient: openSearchObservabilityClient,`. For notebooks you can add your API endpoints in the common `OpenSearchObservabilityPlugin`